### PR TITLE
Area

### PR DESCRIPTION
**Pull request by msurmava**

### CODE REVIEW

File: area/triangle.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#calculates area of a triangle by base and height
+
+class Triangle
```
[the remark this answers is not in the document]
Comment (cyberik-dev) [reply area/triangle.rb:3]: add `class AttributeError < StandardError; end`

File: area/triangle.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  def validate_parameters
+    if @base <= 0 || @height  <=  0
+      warn "Parameters must be positive numbers!"
+      exit 1
+    end
+  end
+
+  def area
+    validate_parameters
+    @base * @height * AREA
```
Comment (cyberik-dev) [inline area/triangle.rb:20]: @base * @height / DIVIDER

**Review by cyberik-dev — CHANGES_REQUESTED** [review]

File: area/triangle.rb (hunk 2 above)
Comment (msurmava) [reply area/triangle.rb]: done

File: area/triangle.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#calculates area of a triangle by base and height
+
+class Triangle
+  AREA = 1/2.to_f
+
+  def initialize(base,height)
+    @base = base.to_f
+    @height = height.to_f
+  end
+
```
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb:11]: done

File: area/triangle.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#calculates area of a triangle by base and height
+
+class Triangle
+  AREA = 1/2.to_f
```
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb]: done

File: area/triangle.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#calculates area of a triangle by base and height
```
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb]: done

File: area/triangle.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb:11]: done

File: area/triangle.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+class Triangle
+  AREA = 1/2.to_f
+
+  def initialize(base,height)
+    @base = base.to_f
+    @height = height.to_f
+  end
+
+  def validate_parameters
```
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb:21]: done

File: area/triangle.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    if @base <= 0 || @height  <=  0
+      warn "Parameters must be positive numbers!"
+      exit 1
+    end
+  end
+
+  def area
```
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb:27]: done

File: area/triangle.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb]: done

File: area/triangle.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+class Triangle
+  AREA = 1/2.to_f
+
+  def initialize(base,height)
+    @base = base.to_f
+    @height = height.to_f
+  end
+
+  def validate_parameters
+    if @base <= 0 || @height  <=  0
```
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb]: done

File: area/triangle.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (msurmava) [reply area/triangle.rb]: done

File: area/triangleareacalculator.rb [hunk 9]
```diff
@@ -0,0 +1,36 @@
+# calculates area of a triangle by base and height
```
Comment (cyberik-dev) [inline area/triangleareacalculator.rb:1]: good job, but fix rubocop remarks please

> 1 file inspected, 3 offenses detected, 3 offenses auto-correctable

**Pull request merged**